Studio-Lovelies/GG-JointJustice-Unity
# Feature 56 Optional Target Actor for SetPose and SetEmotion

## Summary
You can now call `&SET_POSE:<pose_name>,<target_actor>` to set the pose of a target actor. By default you'll generally want `&SET_POSE:<pose_name>` because that uses the current actor which is what you want 99% of the time. We support the same functionality for SetEmotion.

To do this I created a lookup table on ActorController that corresponds `ActorData`s to `Actor`s. This assumes that this relationship is 1 ActorData to 1 Actor, which it technically might not be but Bas said we will probably keep them 1:1.

The unfortunate way this works is we have to store a "pending animation" on the target actor because the animator does not work while the actor is disabled (ie: the actor's scene is not visible)

## Before/after screenshots and/or animated gif
(old gif is in accurate removed so it's not misleading)

## Testing instructions (Updated!!)
Load `OffScreenPose - TestScene`
This entire scene just shows Ross the whole time, but if you switch to the scene view you can see things happen o f f   s c r e e n
When the dialogue says Arin is "big now" switch to the Scene view and pan over to see Arin is big.
Similarly when the dialogue says tutorial boy is angry, switch to scene view to see his expression is now angry.

## Additional information

- `[ ]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
closes #56
